### ISSUE TITLE
False positive triggering when uploading files with Cyrillic characters in the name

### ISSUE THREAD

**Issue by flexstat** [coreruleset/coreruleset#3116]
I'm trying to upload a file that contains Cyrillic characters, the rule id:920120 is triggered

### Description

It doesn't matter what the file name is, if there is at least one Cyrillic character, the server returns a 403 error

for example, let it be имяфайла.txt

``
ModSecurity: Warning. Matched "Operator `Rx' with parameter `(?i)^(?:&(?:(?:[acegiln-or-suz]acut|[aeiou]grav|[ain-o]tild)e|[c-elnr-tz]caron|(?:[cgk-lnr-t]cedi|[aeiouy]um)l|[aceg-josuwy]circ|[au]ring|a(?:mp|pos)|nbsp|oslash);|[^\"';=])*$' against variable `FILES:&#1080;.txt' (Value: `&#1080;.txt' ) [file "/opt/bunkerweb/core/modsecurity/files/coreruleset/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "98"] [id "920120"] [rev ""] [msg "Attempted multipart/form-data bypass"] [data "&#1080;.txt"] [severity "2"] [ver "OWASP_CRS/3.3.2"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/210/272"] [hostname "172.30.0.2"] [uri "/upload.php"] [unique_id "167502782964.429571"] [ref "v1091,11t:urlDecodeUnio0,4"], client: 172.30.0.3, server: www.example.com, request: "POST /upload.php HTTP/1.1", host: "dsz.hs", referrer: "dsz.hs"
``

``
ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `5' ) [file "/opt/bunkerweb/core/modsecurity/files/coreruleset/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "80"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "2"] [ver "OWASP_CRS/3.3.2"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"] [hostname "172.30.0.2"] [uri "/upload.php"] [unique_id "167502782964.429571"] [ref ""], client: 172.30.0.3, server: www.example.com, request: "POST /upload.php HTTP/1.1", host: "dsz.hs", referrer: "dsz.hs"
``

i try change rule , add this lines in REQUEST-920-PROTOCOL-ENFORCEMENT.conf

```
SecRule FILES|FILES_NAMES "!@rx (?i)^(?:&(?:(?:[acegiln-or-suz]acut|[aeiou]grav|[ain-o]tild)e|[c-elnr-tz]caron|(?:[cgk-lnr-t]cedi|[aeiouy]um)l|[aceg-josuwy]circ|[au]ring|a(?:mp|pos)|nbsp|oslash);|[^\"';=])*$" \
 "id:920120,\
    phase:2,\
    block,\
    t:none,t:urlDecodeUni,\
    msg:'Attempted multipart/form-data bypass',\
    logdata:'%{MATCHED_VAR}',\
    tag:'application-multi',\
    tag:'language-multi',\
    tag:'platform-multi',\
    tag:'attack-protocol',\
    tag:'paranoia-level/1',\
    tag:'OWASP_CRS',\
    tag:'capec/1000/210/272',\
    ver:'OWASP_CRS/3.3.2',\
    severity:'CRITICAL',\
    setvar:'tx.anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
```

but it didn't work

help solve this issue

### Your Environment

* CRS version (e.g., v3.2.0): 3.3.2
* Paranoia level setting: PL1

**Comment by azurit** [comment]
Hi @flexstat, can you upload full log from this request? Thank you.

**Comment by flexstat** [comment]
> Hi @flexstat, can you upload full audit log related to this request? Thank you.

Hi, 

cat modsec_audit.log


2023/01/30 12:54:00 [warn] 584#584: *329 ModSecurity: Warning. Matched "Operator `Rx' with parameter `(?i)^(?:&(?:(?:[acegiln-or-suz]acut|[aeiou]grav|[ain-o]tild)e|[c-elnr-tz]caron|(?:[cgk-lnr-t]cedi|[aeiouy]um)l|[aceg-josuwy]circ|[au]ring|a(?:mp|pos)|nbsp|oslash);|[^\"';=])*$' against variable `FILES:&#1080;.txt' (Value: `&#1080;.txt' ) [file "/opt/bunkerweb/core/modsecurity/files/coreruleset/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "98"] [id "920120"] [rev ""] [msg "Attempted multipart/form-data bypass"] [data "&#1080;.txt"] [severity "2"] [ver "OWASP_CRS/3.3.2"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/210/272"] [hostname "172.21.0.3"] [uri "/upload.php"] [unique_id "16750832408.740158"] [ref "v1093,11t:urlDecodeUnio0,4"], client: 172.21.0.2, server: www.example.com, request: "POST /upload.php HTTP/1.1", host: "dsz.hs", referrer: "http://dsz.hs/"

**Comment by azurit** [comment]
Sorry but i need an `audit` log, see modsecurity directives:
```
SecAuditEngine
SecAuditLogRelevantStatus
SecAuditLogParts
SecAuditLogType
SecAuditLogStorageDir
SecAuditLogDirMode
SecAuditLogFormat
```
Alternatively, you can provide me with `curl` call which triggers this problem. Thank you.

**Comment by flexstat** [comment]
> Sorry but i need an `audit` log, see modsecurity directives:

where can I find this log file ? 

How can I do this via curl ? The fact is that I upload a file through the form and when uploading a file with Cyrillic, the specified rule is triggered

**Comment by azurit** [comment]
Look into modsecurity.conf and search for directives which i mentioned above.

`curl` is able to do also file upload.

**Comment by flexstat** [comment]
I solved the problem, it was an incorrect encoding of the form page, I added these lines to the beginning and everything was solved.

<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />

**Comment by azurit** [comment]
Cool, thank you!
